aws/api.ts

```diff
@@ -143,226 +137,12 @@ export class API implements cloud.API {
 export class HttpDeployment extends pulumi.ComponentResource implements cloud.HttpDeployment {
     public routes: Route[];
     public staticRoutes: StaticRoute[];
+
+    public /*out*/ readonly api: x.API;
     public /*out*/ readonly url: pulumi.Output<string>; // the URL for this deployment.
     public /*out*/ readonly customDomainNames: pulumi.Output<string>[]; // any custom domain names.
     public /*out*/ readonly customDomains: aws.apigateway.DomainName[]; // AWS DomainName objects for custom domains.
```

Comment:
Note: all routing functionality moved down to apigateway.x.API.  DomainName logic stays here for now.